pkg/comp-functions/functions/vshnmariadb/mariadb_deploy.go

```diff
@@ -0,0 +1,263 @@
+package vshnmariadb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	xfnproto "github.com/crossplane/function-sdk-go/proto/v1beta1"
+	xhelmbeta1 "github.com/vshn/appcat/v4/apis/helm/release/v1beta1"
+	vshnv1 "github.com/vshn/appcat/v4/apis/vshn/v1"
+	"github.com/vshn/appcat/v4/pkg/common/utils"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/functions/common"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	SLIBucketName = "vshn-test-bucket-for-sli"
+)
+
+// DeployMariadb will add deploy the objects to deploy mariadb
+func DeployMariadb(ctx context.Context, svc *runtime.ServiceRuntime) *xfnproto.Result {
+
+	l := controllerruntime.LoggerFrom(ctx)
+
+	comp := &vshnv1.VSHNMariaDB{}
+	err := svc.GetObservedComposite(comp)
+	if err != nil {
+		err = fmt.Errorf("cannot get observed composite: %w", err)
+		return runtime.NewFatalResult(err)
+	}
+
+	l.Info("Creating namespace for mariadb instance")
+	err = createObjectNamespace(ctx, comp, svc)
+	if err != nil {
+		err = fmt.Errorf("cannot create mariadb namespace: %w", err)
+		return runtime.NewFatalResult(err)
+	}
+
+	l.Info("Creating tls certificate for mariadb instance")
+	err = common.CreateTlsCerts(ctx, comp.GetInstanceNamespace(), comp.GetName(), svc)
+	if err != nil {
+		err = fmt.Errorf("cannot create tls certificate: %w", err)
+		return runtime.NewFatalResult(err)
+	}
+
+	l.Info("Creating helm release for mariadb instance")
+	err = createObjectHelmRelease(ctx, comp, svc)
+	if err != nil {
+		err = fmt.Errorf("cannot create helm release: %w", err)
+		return runtime.NewFatalResult(err)
+	}
+
+	l.Info("Creating network policies mariadb instance")
+	sourceNs := []string{comp.GetClaimNamespace()}
+	if svc.GetBoolFromCompositionConfig("slosEnabled") {
+		sourceNs = append(sourceNs, svc.Config.Data["slosNs"])
+	}
+	err = common.CreateNetworkPolicy(ctx, sourceNs, comp.GetInstanceNamespace(), comp.GetName(), svc)
+	if err != nil {
+		err = fmt.Errorf("cannot create helm release: %w", err)
+		return runtime.NewFatalResult(err)
+	}
+
+	l.Info("Get connection details from secret")
+	err = getConnectionDetails(ctx, comp, svc)
+	if err != nil {
```

Comment:
The current implementation of `getConnectionDetails(ctx, comp, svc)` will never return a non-nil error. So you can remove all of this.

Reply:
Removed it